### PR TITLE
WIP Enhance tests for parallelization

### PR DESCRIPTION
**Pull request by btry**
### Changes description

To speedup tests,,  this PR refactors unit tests to become insensitive to concurrent executions of different tests

### Checklist

Please check if your PR fulfills the following specifications:

- [x] Tests for the changes have been added
- [ ] Docs have been added/updated

### CODE REVIEW

File: inc/agent.class.php [hunk 1]
```diff
@@ -427,7 +427,8 @@ public function prepareInputForAdd($input) {
 
       $maxAgents = $entityConfig->getField('device_limit');
       $entityId = $_SESSION['glpiactive_entity'];
-      $deviceCount = countElementsInTable($this->getTable(), "`entities_id`='$entityId'");
+      $DbUtils = new DbUtils();
+      $deviceCount = $DbUtils->countElementsInTable($this->getTable(), "`entities_id`='$entityId'");
```
Comment (btry) [inline inc/agent.class.php]: code refactor to avoid call to deprecated method

File: inc/common.class.php [hunk 2]
```diff
@@ -140,4 +140,21 @@ public static function recursiveRmdir($dir) {
...
+    * @return NULL|integer
+    */
+   public static function getMax(CommonDBTM $item, $condition, $fieldName) {
+      $rows = $item->find($condition, "`$fieldName` DESC", '1');
+      $line = array_pop($rows);
+      if ($line === null) {
+         return null;
+      }
+      return $line[$fieldName];
+   }
```
Comment (btry) [inline inc/common.class.php]: A helper method I recently created for Formcreator. Useful to easily get the highest existing ID in the MQTT logs table (see refactored test s below)

**Comment by ajsb85** [comment]
Rouge commit message type? xD

**Comment by btry** [comment]
> Rogue commit message type

Yes, I wanted it to NOT follow the convention because this commit will be removed or modified. This is a test. :D

**Comment by DIOHz0r** [comment]
@btry your last change for the agents count limit is merged on develop, could you rebase this PR to check again the tests? thank you.

**Comment by btry** [comment]
@DIOHz0r This is useless. The PR you merged is a commit from this one. 

I think I'll give up this PR, as I don't find a solution to parallelize without breaking things. I'll only pick an other change I did here to make unit tests less sensitive each other, and close it.

**Comment by btry** [comment]
#342 contains test refactor I did to permit parallelization of unit tests. If the branch passes, let's merge and when we can solve the issues with the device limit count parallelization, we are done.

**Comment by btry** [comment]
obsoleted

**Pull request closed without merging**